src/lddc.cpp

```diff
@@ -452,6 +462,109 @@ void Lddc::InitPointcloud2Msg(const StoragePacket& pkg, PointCloud2& cloud, uint
   memcpy(cloud.data.data(), points.data(), pkg.points_num * sizeof(LivoxPointXyzrtlt));
 }
 
+void Lddc::InitPointcloud2NonReturnRaysMsg(const StoragePacket& pkg, PointCloud2& cloud, uint64_t& timestamp, const std::string& frame_id) {
+  InitPointcloud2NonReturnRaysMsgHeader(cloud, frame_id);
+
+  cloud.point_step = sizeof(LivoxPointRtp);
+
+  cloud.is_bigendian = false;
+  cloud.is_dense     = true;
+
+  if (!pkg.points.empty()) {
+    timestamp = pkg.base_time;
+  }
+
+  #ifdef BUILDING_ROS1
+      cloud.header.stamp = ros::Time( timestamp / 1000000000.0);
+  #elif defined BUILDING_ROS2
+      cloud.header.stamp = rclcpp::Time(timestamp);
+  #endif
+
+  std::vector<LivoxPointRtp> rays;
+  size_t i {0};
+  for (i = 0; i < pkg.points_num; ++i) {
+    if (pkg.points[i].range == 0.0)
+    {
+      LivoxPointRtp ray;
+      double src_x = sin(pkg.points[i].theta) * cos(pkg.points[i].phi);
+      double src_y = sin(pkg.points[i].theta) * sin(pkg.points[i].phi);
+      double src_z = cos(pkg.points[i].theta);

```

Comment:
is this a repeat of the x, y, z calculated in `LidarPubHandler::ProcessSphericalPoint`?

Reply:
in `LidarPubHandler::ProcessSphericalPoint` the point.x, .y, .z are computed with the depth from the shot of the lidar. In case of no return the depth is 0, so x, y, z are all 0.0. 
Here I use a fixed depth / radius of 1m to compute the ray's direction vector.